src/overlays/Overlay.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useRef, useId } from "react"
+import { createPortal } from "react-dom"
+import FocusTrap from "focus-trap-react"
+import { faXmark } from "@fortawesome/free-solid-svg-icons"
+import Icon from "../icons/Icon"
+import Heading from "../text/Heading"
+import usePortal from "../hooks/usePortal"
+
+import "./Overlay.scss"
+
+
+export interface OverlayHeaderProps {
+  children: React.ReactNode
+  /** Place the close button at the inline end of the header */
+  closeButtonEnd?: boolean
+  /** Additional class name */
+  className?: string
+}
+
+const OverlayHeader = (props: OverlayHeaderProps) => {
+  const classNames = ["seeds-overlay-header"]
+  if (props.closeButtonEnd) classNames.push("has-close-button-end")
+  if (props.className) classNames.push(props.className)
+
+  const headerRef = useRef<HTMLElement>(null)
+  // Focus on the heading on render so that it is read by screen readers
+  useEffect(() => {
+    headerRef.current?.querySelector<HTMLElement>(".seeds-overlay-heading")?.focus()
+  }, [])
+
+  const closeButton = (
+    <button onClick={() => headerRef.current?.dispatchEvent(new Event("seeds:close", { bubbles: true }))}>
+      <Icon icon={faXmark} size={"lg"} className={"seeds-overlay-close-icon"} />
+    </button>
+  )
+
+  return (
+    <header className={classNames.join(" ")} ref={headerRef}>
+      {!props.closeButtonEnd && closeButton}
+      <Heading priority={1} size="xl" className={"seeds-overlay-heading"} tabIndex={-1}>
```

Comment:
I see its not currently supported but is there any reason why we wouldn't want to support ref on the seeds heading component considering its use in skip links and other screen reader patterns

Reply:
Can you explain what a ref would do to help that? I'm wondering if adding a prop for the heading to use an `id` attribute would accomplish what you're suggesting (or otherwise I'm not understanding the request)

Reply:
Sure! I'm referencing a case like the [ui-c modal](https://github.com/bloom-housing/ui-components/blob/main/src/overlays/Modal.tsx#L33) where the ref is passed to the h1 to receive focus and I'm wondering if supporting that on the seeds <Heading/> component directly could be useful. Is that clarifying?

Reply:
Hey @ColinBuyck, I re-read this and I think I get what you're saying. Would you write up an issue: the `Heading` component about adding this? (I don't think it's in scope for this PR)